omd/packages/nagios/skel/local/lib/nagios/plugins/check_crl

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python3
+# Copyright (C) 2013 - Remy van Elst
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Mark Ruys <mark.ruys@peercode.nl> - 2015-8-27
+# Changelog: - catch openssl parsing errors
+#            - clean up temporary file on error
+#            - add support for PEM CRL's
+#            - fix message when CRL has been expired
+#            - pretty print duration
+
+# Jeroen Nijhof <jnijhof@digidentity.eu>
+# Changelog: - fixed timezone bug by comparing GMT with GMT
+#            - changed hours to minutes for even more precision
+
+# Remy van Elst - raymii.org - 2012
+# 05.11.2012
+# Changelog: - check with hours instead of dates for more precision,
+#            - URL errors are now also catched as nagios exit code.
+
+# Michele Baldessari - Leitner Technologies - 2011
+# 23.08.2011
+
+import time
+import datetime
+import getopt
+import os
+import pprint
+import subprocess
+import sys
+import tempfile
+import urllib.request, urllib.parse, urllib.error
+
+def check_crl(url, warn, crit):
+    tmpcrl = tempfile.mktemp(".crl")
+    #request = urllib.request.urlretrieve(url, tmpcrl)
+    try:
+        urllib.request.urlretrieve(url, tmpcrl)
+    except:
+        print ("CRITICAL: CRL could not be retrieved: %s" % url)
+        os.remove(tmpcrl)
+        sys.exit(2)
+
+    try:
+        inform = 'DER'
+        crlfile = open(tmpcrl, "r")
+        try:
```

Comment:
Do you know why and when this is necessary? Probably, this PR is related:
https://github.com/RaymiiOrg/nagios/pull/6
I am not saying that you should include this PR here, I am just trying to understand under what conditions this will fail and if this could be avoided by using for example `open(..., mode="rb")` or something.

Reply:
Here, we still need some clarification

Reply:
Here, we still need some clarification